object-store/src/test/java/org/hypertrace/config/objectstore/DefaultObjectStoreTest.java

```diff
@@ -115,38 +127,65 @@ void generatesConfigUpsertRequest() {
                 .setUpdateTimestamp(TEST_UPDATE_TIMESTAMP.toEpochMilli())
                 .setConfig(Values.of("updated"))
                 .build());
-    assertEquals(
+    ConfigObject configObject =
         new ConfigObjectImpl<>(
-            new TestObject("updated"), TEST_CREATE_TIMESTAMP, TEST_UPDATE_TIMESTAMP),
-        this.store.upsertObject(this.mockRequestContext, new TestObject("updated")));
+            new TestInternalObject("updated"), TEST_CREATE_TIMESTAMP, TEST_UPDATE_TIMESTAMP);
+    assertEquals(
+        configObject,
+        this.store.upsertObject(this.mockRequestContext, new TestInternalObject("updated")));
     verify(this.mockStub, times(1))
         .upsertConfig(
             UpsertConfigRequest.newBuilder()
                 .setResourceName(TEST_RESOURCE_NAME)
                 .setResourceNamespace(TEST_RESOURCE_NAMESPACE)
                 .setConfig(Values.of("updated"))
                 .build());
+    verify(this.configChangeEventGenerator, times(1))
+        .sendCreateNotification(
+            eq(this.mockRequestContext),
+            eq(TestApiObject.class.getName()),
+            eq(
+                Value.newBuilder()
+                    .setStructValue(Struct.newBuilder().putFields("api_name", Values.of("updated")))
+                    .build()));
   }
 
   @lombok.Value
-  private static class TestObject {
+  private static class TestInternalObject {
```

Comment:
nit : The test seems to suggest that object persisted in DB is completely different from object being published as part of change event. That is certainly possible. However in most of our usecases, the object being persisted will encapsulate the object being published as part of change event.